### ISSUE TITLE
No exception handling for commit-msg hook (gitlint 0.14.0)

### ISSUE THREAD

**Issue by BrunIF** [jorisroovers/gitlint#166]
install gitlint

```bash
pip install gitlint
....
gitlint generate-config
gitlint install-hook
git add .
git commit -m "qw"
```

Error:
```bash
gitlint: checking commit message...
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/gitlint/git.py", line 46, in _git
    result = sh.git(*command_parts, **git_kwargs)  # pylint: disable=unexpected-keyword-arg
  File "/usr/local/lib/python3.9/site-packages/sh.py", line 1427, in __call__
    return RunningCommand(cmd, call_args, stdin, stdout, stderr)
  File "/usr/local/lib/python3.9/site-packages/sh.py", line 774, in __init__
    self.wait()
  File "/usr/local/lib/python3.9/site-packages/sh.py", line 792, in wait
    self.handle_command_exit_code(exit_code)
  File "/usr/local/lib/python3.9/site-packages/sh.py", line 815, in handle_command_exit_code
    raise exc
sh.ErrorReturnCode_128:

  RAN: /usr/local/Cellar/git/2.29.0/libexec/git-core/git rev-parse --abbrev-ref HEAD

  STDOUT:
HEAD


  STDERR:
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/gitlint", line 8, in <module>
    sys.exit(cli())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/gitlint/cli.py", line 348, in run_hook
    ctx.invoke(lint)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/gitlint/cli.py", line 291, in lint
    violations = linter.lint(commit)
  File "/usr/local/lib/python3.9/site-packages/gitlint/lint.py", line 73, in lint
    LOG.debug("Commit Object\n" + ustr(commit))
  File "/usr/local/lib/python3.9/site-packages/gitlint/utils.py", line 111, in ustr
    return str(obj)
  File "/usr/local/lib/python3.9/site-packages/gitlint/git.py", line 187, in __str__
    return sstr(self.__unicode__())  # pragma: no cover
  File "/usr/local/lib/python3.9/site-packages/gitlint/git.py", line 184, in __unicode__
    return args[0]._try_cache(cache_decorator.cachekey, cache_func_result)
  File "/usr/local/lib/python3.9/site-packages/gitlint/cache.py", line 12, in _try_cache
    cache_populate_func()
  File "/usr/local/lib/python3.9/site-packages/gitlint/cache.py", line 43, in cache_func_result
    args[0]._cache[cache_decorator.cachekey] = func(*args)
  File "/usr/local/lib/python3.9/site-packages/gitlint/git.py", line 334, in branches
    return [self.context.current_branch]
  File "/usr/local/lib/python3.9/site-packages/gitlint/cache.py", line 44, in wrapped
    return args[0]._try_cache(cache_decorator.cachekey, cache_func_result)
  File "/usr/local/lib/python3.9/site-packages/gitlint/cache.py", line 12, in _try_cache
    cache_populate_func()
  File "/usr/local/lib/python3.9/site-packages/gitlint/cache.py", line 43, in cache_func_result
    args[0]._cache[cache_decorator.cachekey] = func(*args)
  File "/usr/local/lib/python3.9/site-packages/gitlint/git.py", line 359, in current_branch
    current_branch = ustr(_git("rev-parse", "--abbrev-ref", "HEAD", _cwd=self.repository_path)).strip()
  File "/usr/local/lib/python3.9/site-packages/gitlint/git.py", line 64, in _git
    raise GitContextError(u"Current branch has no commits. Gitlint requires at least one commit to function.")
gitlint.git.GitContextError: Current branch has no commits. Gitlint requires at least one commit to function.
```

**Comment by sigmavirus24** [comment]
The problem here is:

```
sh.ErrorReturnCode_128:

  RAN: /usr/local/Cellar/git/2.29.0/libexec/git-core/git rev-parse --abbrev-ref HEAD

  STDOUT:
HEAD


  STDERR:
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

In other words, gitlint is shelling out to git and `git` is erroring. Specifically it's running `git rev-parse --abbrev-ref HEAD` which seems to fail in your case. Can you explain why that happens? Can you replicate that?

**Comment by BrunIF** [comment]
```bash
➜ ~/Development/git-lint git:(main) ✗ git log
fatal: your current branch 'main' does not have any commits yet
```
My repository is empty now. 

```bash
➜ ~/Development/git-lint git:(main) ✗ git rev-parse --abbrev-ref HEAD
HEAD
fatal: ambiguous argument 'HEAD': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
```

I deleted `commit-msg` hooks. Create the first commit. Returned this file back. The second commit worked normally.

```bash
➜ ~/Development/git-lint git:(main) ✗ git commit -am "test 2"
gitlint: checking commit message...
3: B6 Body message is missing
-----------------------------------------------
gitlint: Your commit message contains the above violations.
Continue with commit anyways (this keeps the current commit message)? [y(es)/n(no)/e(dit)]
```

Need fix when the repository is empty.

**Comment by jorisroovers** [comment]
Hi, 

thanks for opening this issue, there's indeed a bug here related to exception handling.

If you look at the last line in your error message, you'll see that gitlint already detects the empty repository scenario:

```
gitlint.git.GitContextError: Current branch has no commits. Gitlint requires at least one commit to function.
```

It's just that the exception isn't properly handled. This is because I overlooked the exception handling in the commit-msg hook when porting it to python in the last release (0.14.0). I'm a bit surprised testing didn't catch it 🤷‍♂️

What needs to happen here, is adding exception handling in the `run_hook` method:

https://github.com/jorisroovers/gitlint/blob/92c7c4af862ff65dc9f0d5b24b4839df51582dd8/gitlint/cli.py#L341

Basically identical to what is done here:

https://github.com/jorisroovers/gitlint/blob/92c7c4af862ff65dc9f0d5b24b4839df51582dd8/gitlint/cli.py#L241-L249

You'll notice that if you just run `gitlint` in an empty repo, that the exception is correctly handled and the error is displayed normally.

I'll happily accept a PR (note that it needs to include a test case for this particular scenario), or otherwise fix this myself later :-)

**Comment by jorisroovers** [comment]
Fixed on the `main` branch. This will go out as part of 0.15.0 (#160), which is soon.
